### ISSUE TITLE
Should contract output never be merged?

### ISSUE THREAD

**Issue by nemo83** [bloxbean/cardano-client-lib#440]
More a question than a bug.

I just tried to send to the same script address 10 utxo in one tx. Outputs have been merged by default. Which I didn't think/expect.

Should output sent to script never be merged?

Some options,
* leave it as is (but dev will learn the hard way!)
* never merge pay to contract address, and user has no control
* pay to contract accepts a flag allowMerge (set to false by default)
* update merge outputs method from being true/false, to accept an enum: mergeOutput(NO|PAYMENT_ADDRESS_ONLY|ALL)

**Comment by satran004** [comment]
Yes, currently the only way to achieve that is through QuickTxBuilder's `mergeOutput(false)`. 

In the case of `mergeOutput = true`, which is the default behavior, if there are multiple script outputs with the same script address, the values will be merged into one output, and only the first datum will be used.

But you are right. I think we should not allow output merging for scripts, as in this case, the user's intention is to create multiple outputs, possibly with different datums.

One way to achieve this would be to prevent merging if the output contains any datum or script reference.

So, only outputs with values will be allowed to merge. Alternatively, we can also detect if it's a script address.

Let's tag this as an enhancement.
